src/Microsoft.AspNetCore.Mvc.Analyzers/TopLevelParameterNameAnalyzer.cs

```diff
@@ -144,6 +150,35 @@ internal static string GetName(in SymbolCache symbolCache, ISymbol symbol)
             return symbol.Name;
         }
 
+        internal static bool SpecifiesModelType(in SymbolCache symbolCache, IParameterSymbol parameterSymbol)
+        {
+            foreach (var attribute in parameterSymbol.GetAttributes(symbolCache.IBinderTypeProviderMetadata))
+            {
+                for (var i = 0; i < attribute.NamedArguments.Length; i++)
```

Comment:
It would be good to comment these with examples of what you're trying to match on.